x/group/types.go

```diff
@@ -24,38 +24,45 @@ type DecisionPolicyResult struct {
 type DecisionPolicy interface {
 	codec.ProtoMarshaler
 
+	// GetVotingPeriod returns the duration after proposal submission where
+	// votes are accepted.
+	GetVotingPeriod() time.Duration
+	// GetMinExecutionPeriod returns the minimum duration after proposal
+	// submision from which the proposal can be executed. It can return 0.
+	GetMinExecutionPeriod() time.Duration
```

Comment:
Hmm personally this seems like the wrong abstraction. I would like to hear arguments about why it's better to put these parameters in the decision policy rather than in the group itself. It just feels like duplication. The only thing I think a decision policy should have control of is at what point in time do we calculate the weights i.e. start of the proposal, end of the proposal, in between.

Reply:
it allows a same group to define different decision policies for different use cases

Reply:
But a decision policy is per group account right? Why can't this data be stored at the group account level?

Reply:
Every decision policy is going to have these two fields anyway. I can't see a case where you might calculate the voting period and min_execution_period some other way